docs/overview.md

```diff
@@ -1,26 +1,116 @@
+
+## What is the MyST AST and Specification?
+
+The {term}`MyST AST` is a version of parsed MyST content that follows a specific structure so that Renderers can convert into different outputs. MyST AST conforms to the {term}`MyST Specification`. The {term}`MyST Specification` is a superset of [the `mdast` specification](https://github.com/syntax-tree/mdast) (meaning that MyST includes all of the syntax that `mdast` supports, plus some extra ones like Directives).
+
+Here's an example of MyST content, and the same content parsed as MyST AST:
+
+````{list-table}
+:header-rows: 1
+- - Content
+  - AST
+- - `**Hello there!**`
+  - ```yaml
+    - type: block
+    children:
+      - type: paragraph
+        children:
+          - type: strong
+            children:
+              - type: text
+                value: Hello there!
+    ```
+````
+
+The easiest way to understand how {term}`MyST Markdown` gets converted to {term}`MyST AST` is to [explore the MyST Sandbox](https://mystmd.org/sandbox). Click the {kbd}`AST` tab to see the underlying AST structure for anything that you type. The {kbd}`PRE` tab represents the initial parsing phase, and the {kbd}`POST` tab represents the AST after the resolving phase.
+
+(overview-transformers)=
+## How do MyST Transformers work?
+
+MyST Transformers are a way to convert a {term}`MyST AST` node into another type of node. Transformers operate on AST rather than on raw Markdown because AST has more standardized structure to work with.
+
+For example, consider a Markdown link like `[some text](#a-label)`. In MyST Markdown, this defines a **cross-reference** to `#a-label`, but it uses Markdown link syntax. We use a MyST Transformer to convert that Markdown to a cross-reference AST node like so:
+
+- First parse the Markdown `[some text](#a-label)`.
+- The result is a MyST AST node for a **Markdown link**.
+- Next, search the document AST for any Markdown link nodes with a target that starts with `#`. Check if there's [a target](./cross-references.md) for the link. If so, it is meant to be a cross reference.
+- If so, run a **Transformer** that converts the Markdown Link node into a Cross Reference node.
+
+Some other uses for Transformers include:
+
+- Lifting metadata from `code-cells` to their parent structures
+- Check that figures have alt-texts
+- Convert non-standard AST nodes (e.g., ones generated from a custom user directive) into ones that MyST knows how to render[^ex-transform-node].
+
+[^ex-transform-node]: This is a pattern used in e.g. https://github.com/projectpythia-mystmd/cookbook-gallery/blob/main/pythia-gallery.py where an `executable transform` (non-JS transform that communicates over `STDIO` with `JSON`) takes custom `pythia-cookbooks` nodes and converts them (via some HTTP fetches) to a grid of cards by outputting the relevant grid and card AST nodes.
```

Comment:
How do you communicate with JSON? Or is it communicating *in* JSON with .... the Python script?